scipy/stats/tests/test_morestats.py

```diff
@@ -2575,9 +2575,13 @@ def test_directionalmean_correctness(self):
         reference_mean = np.array([0.298, -0.135, -0.945])
         assert_allclose(mean_rounded, reference_mean)
 
-    def test_directionalmean_2d(self):
-        # Test that for circular data directionalmean yields
-        # same result as circmean
+        expected_var = 0.025335389565304012
+        directional_var = stats.directionalvar(data)
```

Comment:
I found the reference values for the mean in Fisher's article:
![image](https://user-images.githubusercontent.com/6570539/183197918-ed22021e-3d41-4b40-bfbc-a5062f2dcc1e.png)

I don't have Mardia and Jupp's book. Can you show how it appears in the reference? And how do they describe the directional variance in the reference?

Reply:
Checking again, Mardian and Jupp actually do not define a general `directional variance`. I had just assumed generalizing the formula for circular variance to n dimensions would make sense.
There is one section about variance on the $S^2$ sphere which denoes $2(1-\hat{R})$ as spherical variance (see below). In my opinion, the $1-\hat{R}$ definition is fine as long as it is documented.
<img width="431" alt="image" src="https://user-images.githubusercontent.com/40656107/184479590-d8b4a276-7ed9-4f92-b957-6d9d0433ea49.png">

Reply:
SciPy doesn't have a good track record in this department. For years, there was no reference for our implementations of circular standard deviation or variance (gh-10096) and the definition of one of them was inconsistent with the literature (gh-5747). I would prefer not to risk that again. I think we need to use a definition that is consistent with the literature and cite our source. If there are no references for higher dimensions, we might need to limit this to 3D. 
Speaking of which, was our reference for the mean valid in higher dimensions?

Reply:
I see your point not to repeat mistakes of the past.
Regarding `directionalmean`:
Mardia and Jupp define the mean direction for arbitrary dimensions (p. 163 in the 2000 edition) so we are safe.
<img width="447" alt="image" src="https://user-images.githubusercontent.com/40656107/184502405-e0236989-4de5-4dd5-9ae1-e53f474fbb48.png">

Acc. to [this thesis](https://ses.library.usyd.edu.au/bitstream/handle/2123/1132/03chapter2.pdf;jsessionid=3FEF169F7A56C8E246F95D8C05293673?sequence=3), Mardia (1972) defines the spherical varience in the same way as for the circular case. Cannot find this in one of Mardia's 1972 articles though.

Reply:
One more data point for spherical variance: [this paper](https://www.witpress.com/Secure/elibrary/papers/AIR08/AIR08039FU1.pdf) derives that the $1-|R|$ corresponds to the median absolute bias. I have to admit at this point I am not sure how to proceed. Most literature uses $1-|R|$ but there is no general concensus from what I see. Maybe get some more opinions here?

Reply:
It could be called "directional_dispersion" (I think we'd have to add underscores... there is still time to change to `directional_mean`), and there could be a parameter to select the desired version.

Reply:
@rkern it sounded like you were interested in seeing directional statistics in `scipy.stats`. Do you have an opinion about what we should do with this function? It was supposed to be an n-dimensions directional variance, but it sounds like there is not consensus in the literature about what that means. Some sources use $1 - |R|$ in 3D, another uses $2 ( 1 - |R|)$ in 3D, and none seem to generalize to n-D.

Reply:
I don't care that much about it. It doesn't seem like there's a dominant definition, and all of them are easily computed from the directional mean result, so I'm comfortable leaving this function out.

Reply:
Computing both of the $2(1-|R|)$ and $1-|R|^2$ variants from Mardia and Jupp in the `directionalmean()` `Examples` would be a good replacement. By the way, those are derived in that chapter for general (hyper)spheres, not just 3D spheres.

Reply:
Usually I'd prefer to leave out functions that are simple for a user, but I'm conflicted here. @dschmitz89 I'd suggest sending an email (if you haven't already) to get opinions, as you'd need to do that before we merge, anyway.

Reply:
summer heat and life keep me distracted from scipy a bit at the moment.

From what I understood we tend to:
* rename `directionalmean` to `directional_mean`
* delete this PR as there is no canonical definition of directional variance
* add examples for directional variance to the `directional_mean` docstring

Will also send on the mailing list.

Reply:
Hmm that's not what I had in mind.
* I only suggested renaming `directionalmean` to `directional_mean` _if_ we were going to rename this PR to `directional_dispersion`. That was just because I thought `directionaldispersion` was too long.
* I did not suggest deleting this PR; only Robert did. I suggested that you ask for opinions from the mailing list about whether to add it or not.

I got the Mardia and Jupp book, and they definitely call $2(1-R)$ the "spherical variance" for any number of dimensions. I wouldn't mind having an implementation of this quantity.

If you decide not to complete this PR, then perhaps `directionalmean` should return $R$, too. Then the user can define whatever measure of dispersion that they prefer, whether it's $1-R$, $1-R^2$, or $2 (1-R)$.

Reply:
If we call this `directional_variance` with the $2(1-R)$ definition, I suggest to add a big warning in the docs that the definition is different from circvar.

I like the idea of adding $R$ to the output of `directionalmean`. Would definitely avoid future issues of users asking why their favourite definition of the directional variance is not used under the hood.

Reply:
Yeah. I'm liking that. If we can come up with a new name for `directionalmean` that suggests it might return more than just the mean, I think we could do away with the need for a `directionalvar`.